paperqa/settings.py

```diff
@@ -330,6 +330,20 @@ class IndexSettings(BaseModel):
         ),
     )
 
+    def get_named_index_directory(self) -> anyio.Path:
+        """Get the directory where the index, when named, will be located.
+
+        Raises:
+            ValueError: If the index name was unset, because otherwise the name is
+                autogenerated.
+        """
+        if self.name is None:
+            raise ValueError(
+                "Getting a named index directory requires an index name to have been"
+                " specified, please specify a name."
+            )
+        return anyio.Path(self.index_directory) / self.name
```

Comment:
Why is this a `anyio.Path` and `index_directory` is `os.PathLike`? Are those types compatible?

Reply:
Yeah good question, it comes down to "be vague in what you accept, and specific in what you return"

`index_directory`  is an input that allows general types of `str | os.PathLike`:
- `str` is for places like configs
- `os.PathLike` is basically an `ABC` for `pathlib.Path`-like objects

The return is a concretion of `os.PathLike`, which is an `anyio.Path`. We use `anyio.Path` so we can do `async` operations with the `Path`